src/controller/tests/data_model/TestRead.cpp

```diff
@@ -1423,6 +1424,67 @@ void TestReadInteraction::TestReadAttributeTimeout(nlTestSuite * apSuite, void *
     NL_TEST_ASSERT(apSuite, ctx.GetExchangeManager().GetNumActiveExchanges() == 0);
 }
 
+void TestReadInteraction::TestSubscribeAttributeTimeout(nlTestSuite * apSuite, void * apContext)
```

Comment:
Please provide a description above this test case describing what it's testing, and how it's doing so.

Reply:
After client initiated subscription request, test expire session so that subscription fails to establish, and trigger the timeout error.
Client would automatically try to resubscribe and bump the value for numResubscriptionAttemptedCalls.

updated the comments in code. thanks